### PR TITLE
PHPStan workflow + PHPStan fixes

### PR DESCRIPTION
**Pull request by DjordyKoert**
I noticed that PHPStan was available but not actually being used. 

This PR contains a few changes & fixes, namely:
- PHPStan is now ran in CI
- Added PHPStan script (running PHPStan locally)
- A bunch of fixes that I encountered while running PHPStan locally
	- I was not able to fix some though because I don't have the required knowledge of these parts (or maybe some of these are actually bugged currently) . These should be added to baseline or fixed if someone knows how to.

```txt
 ------ -------------------------------------------------------------------------------------- 
  Line   Filament/Resources/UserResource/Pages/EditProfile.php                                 
 ------ -------------------------------------------------------------------------------------- 
  142    Access to an undefined property chillerlan\QRCode\QROptions::$svgLogo.                
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property  
  143    Access to an undefined property chillerlan\QRCode\QROptions::$svgLogoScale.           
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property  
  147    Access to protected property chillerlan\QRCode\QROptions::$version.                   
  149    Access to protected property chillerlan\QRCode\QROptions::$outputBase64.              
  150    Access to protected property chillerlan\QRCode\QROptions::$eccLevel.                  
  151    Access to protected property chillerlan\QRCode\QROptions::$addQuietzone.              
  153    Access to protected property chillerlan\QRCode\QROptions::$connectPaths.              
  154    Access to protected property chillerlan\QRCode\QROptions::$drawCircularModules.       
  157    Access to protected property chillerlan\QRCode\QROptions::$svgDefs.                   
 ------ -------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   Models/Node.php                                                                       
 ------ -------------------------------------------------------------------------------------- 
  242    Access to an undefined property App\Models\Node::$servers_sum_memory.                 
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property  
  249    Access to an undefined property App\Models\Node::$servers_sum_disk.                   
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property  
  256    Access to an undefined property App\Models\Node::$servers_sum_cpu.                    
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property  
 ------ -------------------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Services/Deployment/FindViableNodesService.php                                                                                                                 
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  30     Parameter #1 $value of function collect expects Illuminate\Contracts\Support\Arrayable<(int|string), mixed>|iterable<(int|string), mixed>|null, string given.  
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------- 

                                                                                                                        
 [ERROR] Found 13 errors    
```

### CODE REVIEW

**Comment by DjordyKoert** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by lancepioch** [comment]
Thanks for the work, I'm going to convert this into a draft until it's ready!

File: app/Filament/Resources/NodeResource/RelationManagers/NodesRelationManager.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 namespace App\Filament\Resources\NodeResource\RelationManagers;
 
+use App\Models\Node;
```
Comment (Boy132) [inline app/Filament/Resources/NodeResource/RelationManagers/NodesRelationManager.php:5]: Import not needed
```suggestion
```

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -29,3 +29,26 @@ jobs:
...
+    name: PHPStan
+    runs-on: ubuntu-latest
+    steps:
+      - name: Code Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: "8.2"
```
Comment (Boy132) [inline .github/workflows/lint.yaml:42]:
```suggestion
          php-version: "8.3"
```

File: .github/workflows/lint.yaml [hunk 3]
```diff
@@ -29,3 +29,26 @@ jobs:
 
       - name: Pint
         run: vendor/bin/pint --test
+  phpstan:
```
Comment (Boy132) [inline .github/workflows/lint.yaml:32]: Can you also rename the existing "Lint" job to "Pint" and change the php to 8.3 too?

**Review by Boy132 — CHANGES_REQUESTED** [review]

**Comment by Boy132** [comment]
> @Boy132 I see that you have opened #343 to fix the remaining issues. Should the current errors be added to a phpstan baseline or will this be merged with a failing PHPStan job?

We will properly merge #343 first and then your PR. Then all phpstan errors should be fixed.

**Review by Boy132 — APPROVED** [review]
LGTM

File: composer.json [hunk 4]
```diff
@@ -70,6 +70,7 @@
     "scripts": {
         "cs:fix": "php-cs-fixer fix",
         "cs:check": "php-cs-fixer fix --dry-run --diff --verbose",
+        "phpstan": "phpstan --memory-limit=1G",
```
Comment (lancepioch) [inline composer.json:73]:
```suggestion
        "phpstan": "phpstan --memory-limit=-1",
```

File: app/Repositories/Daemon/DaemonServerRepository.php [hunk 5]
```diff
@@ -25,9 +25,9 @@ public function getDetails(): array
         Assert::isInstanceOf($this->server, Server::class);
 
         try {
-            $response = $this->getHttpClient()->get(
+            return $this->getHttpClient()->get(
                 sprintf('/api/servers/%s', $this->server->uuid)
-            )->throw();
+            )->throw()->json();
```
Comment (lancepioch) [inline app/Repositories/Daemon/DaemonServerRepository.php:30]: Nice, didn't know you could chain that!

File: .github/workflows/lint.yaml [hunk 6]
```diff
@@ -29,3 +29,26 @@ jobs:
...
+          coverage: none
+
+      - name: Setup .env
+        run: cp .env.example .env
+
+      - name: Install dependencies
+        run: composer install --no-interaction --no-progress --prefer-dist
+
+      - name: PHPStan
+        run: vendor/bin/phpstan --memory-limit=1G
```
Comment (lancepioch) [inline .github/workflows/lint.yaml:54]:
```suggestion
        run: vendor/bin/phpstan --memory-limit=-1
```

Comment (DjordyKoert) [reply .github/workflows/lint.yaml]: Why are you removing the memory limit if I may ask?

Comment (lancepioch) [reply .github/workflows/lint.yaml]: Because there's no point in limiting it, especially in GitHub

**Pull request merged**